Models/Assimilation/Assimilation.cs

```diff
@@ -0,0 +1,109 @@
+﻿using DCAPST.Interfaces;
+
+namespace DCAPST
+{
+    public class Assimilation : IAssimilation
+    {
+        public AssimilationType Type { get; set; }
+
+        public IPartialCanopy Partial;
+        public ICanopyParameters Canopy;
+        public IPathwayParameters Path;              
+
+        public double CO2AssimilationRate { get; set; }
+        public double WaterUse { get; set; }
+        public double LeafTemperature { get; set; }
+
+        public double IntercellularCO2 { get; set; }
+        public double MesophyllCO2 { get; set; }
+        public double ChloroplasticCO2 { get; set; }
+        public double ChloroplasticO2 { get; set; }
+
+        public Assimilation(AssimilationType type, IPartialCanopy partial)
+        {
+            Type = type;
+
+            Partial = partial;
+            Canopy = partial.Canopy;
+            Path = partial.Canopy.Pathway;           
+
+            MesophyllCO2 = Canopy.AirCO2 * Path.IntercellularToAirCO2Ratio;
+            ChloroplasticCO2 = MesophyllCO2 + 20;
+            ChloroplasticO2 = 210000;
+        }
+        
+        // Per Leaf
+        public double MesophyllCO2ConductanceAtT => TemperatureFunction.Val(LeafTemperature, Partial.MesophyllCO2Conductance25, Path.MesophyllCO2ConductanceParams);
+
+        public bool TryUpdateAssimilation(ILeafWaterInteraction Water, PhotosynthesisParams Params)
+        {
+            AssimilationCalculator calc;
+
+            // TODO: There might be a better structure for this, since it's only used to
+            // determine the assimilation parameters
+            if (Canopy.Type == CanopyType.C3) 
+                calc = new CalculatorC3(this, Partial);
+            else if (Canopy.Type == CanopyType.C4) 
+                calc = new CalculatorC4(this, Partial);
+            else 
+                calc = new CalculatorCCM(this, Partial);
+
+            var aparam = calc.GetAssimilationParams();
+
```

Comment:
Can you move these calculations into the AssimilationCalculator?
Then you can use a factory method to get the typed Calculator implementation (C3/C4/CCM)
Call CalcLimited() or CalcPotential() and have the similarities encapsulated in the base class with the exceptions being contained in the derived classes.

Reply:
Yeah I can look into something like that. It does smell a bit, I just didn't consider fixing it much of a priority.